OpenEdXMobile/src/main/java/org/edx/mobile/util/DataBindingHelperUtils.kt

```diff
@@ -114,47 +124,121 @@ class DataBindingHelperUtils {
+            if (dateBlock.isToday() && !parentBadgeAdded) {
+                createBadge(textView, CourseDateType.TODAY)
+            }
+            // add date badge at second position OR sub date item other then Today's badge
+            if (hasSameDateTypes(dateBlockItems) && !dateBlock.isDateTypeToday()) {
+                createBadge(textView, dateBlock.dateBlockBadge)
+            }
+        }
+
+        /**
+         * Method to create the Date badge as per given DateType
+         */
+        private fun createBadge(textView: TextView, courseDateType: CourseDateType) {
+            val badgeBackground: Int
+            val textAppearance: Int
+            var badgeStrokeColor: Int = -1
+            var badgeIcon: IconDrawable? = null
+            when (courseDateType) {
+                CourseDateType.TODAY -> {
+                    badgeBackground = R.color.course_today_date
+                    textAppearance = R.style.today_date_badge_text_appearance
+                }
+                CourseDateType.VERIFIED_ONLY -> {
+                    badgeBackground = R.color.black
+                    textAppearance = R.style.verified_only_badge_text_appearance
+                    badgeIcon = IconDrawable(textView.context, FontAwesomeIcons.fa_lock)
+                            .sizeRes(textView.context, R.dimen.small_icon_size)
+                            .colorRes(textView.context, R.color.white)
+                }
+                CourseDateType.COMPLETED -> {
+                    badgeBackground = R.color.tag_light_silver
+                    textAppearance = R.style.completed_badge_text_appearance
+                }
+                CourseDateType.PAST_DUE -> {
+                    badgeBackground = R.color.bullet_light_gray
+                    textAppearance = R.style.past_due_badge_text_appearance
+                }
+                CourseDateType.DUE_NEXT -> {
+                    badgeBackground = R.color.bullet_dark_gray
+                    textAppearance = R.style.due_next_badge_text_appearance
+                }
+                CourseDateType.NOT_YET_RELEASED -> {
+                    badgeBackground = android.R.color.transparent
+                    textAppearance = R.style.not_yet_released_badge_text_appearance
+                    badgeStrokeColor = R.color.tag_light_silver_border
+                }
+                else -> {
+                    return
+                }
+            }
+            addDateBadge(textView, courseDateType.getTitle(), badgeBackground, textAppearance, badgeIcon, badgeStrokeColor)
+        }
+
+        /**
+         * Method to check that all Date Items have same badge status or not
+         *
+         * @return true if all the date items have update badge status else false
+         * */
+        private fun hasSameDateTypes(dateBlockItems: ArrayList<CourseDateBlock>?): Boolean {
+            if (dateBlockItems != null && dateBlockItems.isEmpty().not() && dateBlockItems.size > 1) {
```

Comment:
we can use `dateBlockItems.isNotEmpty()`  instead of `dateBlockItems.isEmpty().not()`

Reply:
replaced.